pkg/api/option.go

```diff
@@ -159,12 +148,27 @@ func WithTimeout(timeout time.Duration) Option {
 	}
 }
 
+// WithHostname sets the X-Machine-Name header to the passed in hostname.
+func WithHostname(hostname string) Option {
+	return func(c *Client) {
+		next := c.doFunc
+		c.doFunc = func(c *Client, req *http.Request) (*http.Response, error) {
+			hostname = url.QueryEscape(strings.TrimSpace(hostname))
+			req.Header.Set("X-Machine-Name", hostname)
+
+			return next(c, req)
+		}
+	}
+}
+
 // WithTimezone sets the TimeZone header to the passed in timezone.
 func WithTimezone(timezone string) Option {
 	return func(c *Client) {
 		next := c.doFunc
 		c.doFunc = func(c *Client, req *http.Request) (*http.Response, error) {
+			timezone = strings.TrimSpace(timezone)
```

Comment:
The idea can be done here. Here's [the piece of code](https://github.com/wakatime/wakatime-cli/blob/develop/cmd/api/api.go#L38-L43) that call `WithTimezone`.

Reply:
What should be done here?

Reply:
Fixed!